applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/server_store.h

```diff
@@ -0,0 +1,413 @@
+#include <zephyr/bluetooth/audio/bap.h>
+#include <zephyr/bluetooth/audio/bap_lc3_preset.h>
+#include <zephyr/bluetooth/addr.h>
+
+struct unicast_server_snk_vars {
+	bool waiting_for_disc;
+	/* PACS response. Location should be a superset of all codec locations. Bitfield */
+	uint32_t locations;
+	/* lc3_preset will propagate to the streams. */
+	struct bt_bap_lc3_preset lc3_preset[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
+	struct bt_audio_codec_cap codec_caps[CONFIG_CODEC_CAP_COUNT_MAX];
+	size_t num_codec_caps;
+	/* One array for discovering the eps */
+	struct bt_bap_ep *eps[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
+	size_t num_eps;
+	enum bt_audio_context supported_ctx;
+	enum bt_audio_context available_ctx;
+	/* We should have all info here. (Locations, stream status etc.) */
+	struct bt_cap_stream cap_streams[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SNK_COUNT];
+};
+
+struct unicast_server_src_vars {
+	bool waiting_for_disc;
+	uint32_t locations;
+	struct bt_bap_lc3_preset lc3_preset[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT];
+	struct bt_audio_codec_cap codec_caps[CONFIG_CODEC_CAP_COUNT_MAX];
+	size_t num_codec_caps;
+	struct bt_bap_ep *eps[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT];
+	size_t num_eps;
+	enum bt_audio_context supported_ctx;
+	enum bt_audio_context available_ctx;
+	struct bt_cap_stream cap_streams[CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT];
+};
+
+/* This struct holds the parameters for a given unicast server/headset device */
+struct server_store {
+	char *name;
+	bt_addr_le_t addr;
+	struct bt_conn *conn;
+	const struct bt_csip_set_coordinator_set_member *member;
+	struct unicast_server_snk_vars snk;
+	struct unicast_server_src_vars src;
+};
+
+struct client_supp_configs {
+	enum bt_audio_codec_cap_freq freq;
+	enum bt_audio_codec_cap_frame_dur dur;
+	enum bt_audio_codec_cap_chan_count chan_count;
+	struct bt_audio_codec_octets_per_codec_frame oct_per_codec_frame;
+};
+
+/** Callback function for bt_bap_unicast_group_foreach_stream().
+ *
+ * @param server The audio server.
+ * @param user_data  User data.
+ *
+ * @retval false Continue iterating.
+ * @retval true Stop iterating.
+ */
+typedef bool (*srv_store_foreach_func_t)(struct server_store *server, void *user_data);
+
+/** Iterate through all stored servers and call the given function for each.
+ *
+ * @note		srv_store_lock() must be called before accessing this function.
+ *
+ * @param func		Function to call for each server.
+ * @param user_data	User data to pass to the function.
+ *
+ * @retval	0 Success.
+ * @retval	-EINVAL if func is NULL.
+ * @retval	-ECANCELED if the iteration was stopped by the callback function.
+ */
+int srv_store_foreach_server(srv_store_foreach_func_t func, void *user_data);
+
+/**
+ * @brief Validate the codec configuration preset.
+ *
+ * @param new		Pointer to the new codec configuration.
+ * @param existing	Pointer to the existing codec configuration.
+ * @param pref_sample_rate_value  Preferred sample rate value.
```

Comment:
[in], [out] in params required: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/dev_model_and_contributions/documentation/styleguide.html#functions